Entities/Natural/Trees/AppleTreeLogic.as

```diff
@@ -0,0 +1,384 @@
+// Apple tree Logic
+
+#include "TreeSync.as"
+
+void onInit(CBlob@ this)
+{
+	InitVars(this);
+	this.server_setTeamNum(-1);
+	TreeVars vars;
+	vars.seed = XORRandom(30000);
+	vars.growth_time = 250 + XORRandom(30);
+	vars.height = 0;
+	vars.max_height = 2 + XORRandom(2);
+	vars.grown_times = 0;
+	vars.max_grow_times = 50;
+	vars.last_grew_time = getGameTime() - 1; //pretend we started a frame ago ;)
+	InitTree(this, vars);
+	this.set("TreeVars", vars);
+	
+	// Produce
+	if (this.hasTag("instant_grow"))
+	{
+		GrowApples(this);
+	}
+}
+
+// Grow the apples
+void GrowApples(CBlob@ this)
+{
+	if (getNet().isServer())
+	{
+		for (int i = 0; i < 2; i++)
+		{	
+			switch (i)
+			{
+				// Apple 1
+				case 0: InitFirstApple(this); break;
+	
+				// Apple 2
+				case 1: InitSecondApple(this); break;

```

Comment:
Why would you do a for loop + switch statement here? Just do `InitFirstApple(this);InitSecondApple(this);`

Reply:
An accidentally-kept remnant of old code. Fixing...